### PR TITLE
Rework Azure Blob Storage documentation to be more clear

### PR DESCRIPTION
**Pull request by XavierGeerinck**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Clarify the documentation for Azure Blob Storage

### CODE REVIEW

File: daprdocs/content/en/operations/components/setup-bindings/supported-bindings/blobstorage.md [hunk 1]
```diff
@@ -40,20 +40,62 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
 
+> Note: by default, a random UUID is generated. See below for Metadata support to set the name
+
 ```json
 {
   "operation": "create",
-  "data": {
-    "field1": "value1"
-  }
+  "data": "YOUR_CONTENT"
```
Comment (orizohar) [inline daprdocs/content/en/operations/components/setup-bindings/supported-bindings/blobstorage.md:48]: As long as we are clarifying, can you please add a comment right after this about the fact that "YOUR_CONTENT" can be different formats (that you show further down)?

**Review by orizohar — DISMISSED** [review]
Thanks for adding more examples and clarifying! Please see my one comment

File: daprdocs/content/en/operations/components/setup-bindings/supported-bindings/blobstorage.md (hunk 1 above)
Comment (Proximus-Azure-Services) [reply daprdocs/content/en/operations/components/setup-bindings/supported-bindings/blobstorage.md:48]: very valid comment! Would love to do so but I am working as well to get it working myself :) the issue I am having here is that a REST endpoint requires a string to be passed (due to the nature of { "data": CONTENT } and it's serialization. 

How do you see this? When working with images for example, I am seeing that I should pass them as Base64 and decode them with an internal function (which I can't seem to get working but that's another discussion)

Comment (XavierGeerinck) [reply daprdocs/content/en/operations/components/setup-bindings/supported-bindings/blobstorage.md:48]: very valid comment! Would love to do so but I am working as well to get it working myself :) the issue I am having here is that a REST endpoint requires a string to be passed (due to the nature of { "data": CONTENT } and it's serialization.

How do you see this? When working with images for example, I am seeing that I should pass them as Base64 and decode them with an internal function (which I can't seem to get working but that's another discussion)

**Review by yaron2 — APPROVED** [review]
lgtm

**Pull request merged**